tests/test_qubit_device_adjoint_jacobian.py

```diff
@@ -39,6 +39,17 @@ def test_not_expval(self, dev):
         with pytest.raises(qml.QuantumFunctionError, match="Adjoint differentiation method does"):
             dev.adjoint_jacobian(tape)
 
+    def test_finite_shots_warns(self):
```

Comment:
It would be nice to have a test with the QNode UI too. Should be quite similar, as this one is already a great case! :)